pkg/protoconv/networkpolicy/k8s.go

```diff
@@ -22,10 +24,27 @@ type KubernetesNetworkPolicyWrap struct {
 
 // ToYaml produces a string holding a JSON formatted yaml for the network policy.
 func (np KubernetesNetworkPolicyWrap) ToYaml() (string, error) {
-	encoder := json.NewYAMLSerializer(json.DefaultMetaFactory, nil, nil)
+	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(np.NetworkPolicy)
+	if err != nil {
+		return "", err
+	}
+	encoder := json.NewSerializerWithOptions(json.DefaultMetaFactory, nil, nil, json.SerializerOptions{
+		Yaml: true,
+	})
+
+	// Kubernetes added a 'status' field for NetworkPolicies in 1.24. See:
+	// * (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#api-change-3)
+	// * (https://github.com/kubernetes/kubernetes/pull/107963)
+	// Using the `NewSerializerWithOptions` from `k8s.io/apimachinery/pkg/runtime/serializer/json` with a `v1.NetworkPolicy` will return the `status` field.
+	// The problem is: Old cluster using kubernetes < 1.24 will fail to apply NetworkPolicies generated this way.
+	// Since the `status` field is not handled by ACS, we delete the field manually if present here.
+	// This code might not be necessary in the future since the feature was withdrawn by the sig-network. See:
+	// * (https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2943-networkpolicy-status#implementation-history)
+	// * (https://github.com/kubernetes/kubernetes/pull/107963#issuecomment-1400220883)
+	delete(uObj, "status")
```

Comment:
All looks correct but I'd suggest to change the order so that comment applies to both Unstructured and removal, and `encoder` stays outside of that.

```go
	// Kubernetes added a 'status' field for NetworkPolicies in 1.24. See:
	// * (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#api-change-3)
	// * (https://github.com/kubernetes/kubernetes/pull/107963)
	// Using the `NewSerializerWithOptions` from `k8s.io/apimachinery/pkg/runtime/serializer/json` with a `v1.NetworkPolicy` will return the `status` field.
	// The problem is: Old cluster using kubernetes < 1.24 will fail to apply NetworkPolicies generated this way.
	// Since the `status` field is not handled by ACS, we delete the field manually if present here.
	// This code might not be necessary in the future since the feature was withdrawn by the sig-network. See:
	// * (https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2943-networkpolicy-status#implementation-history)
	// * (https://github.com/kubernetes/kubernetes/pull/107963#issuecomment-1400220883)
	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(np.NetworkPolicy)
	if err != nil {
		return "", err
	}
	delete(uObj, "status")

	encoder := json.NewSerializerWithOptions(json.DefaultMetaFactory, nil, nil, json.SerializerOptions{
		Yaml: true,
	})
```

Reply:
Makes sense. I've changed the order as you suggested.